aws/rust-runtime/aws-http/src/auth.rs

```diff
@@ -3,27 +3,26 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-use aws_credential_types::provider::{
-    error::CredentialsError, ProvideCredentials, SharedCredentialsProvider,
-};
+use aws_credential_types::cache::{ProvideCachedCredentials, SharedCredentialsCache};
+use aws_credential_types::provider::error::CredentialsError;
 use aws_smithy_http::middleware::AsyncMapRequest;
 use aws_smithy_http::operation::Request;
 use aws_smithy_http::property_bag::PropertyBag;
 use std::future::Future;
 use std::pin::Pin;
 
-/// Sets the credentials provider in the given property bag.
-pub fn set_provider(bag: &mut PropertyBag, provider: SharedCredentialsProvider) {
-    bag.insert(provider);
+/// Sets the credentials cache in the given property bag.
+pub fn set_credentials_cache(bag: &mut PropertyBag, cache: SharedCredentialsCache) {
```

Comment:
This should get called out as its own breaking change, as this is called directly by some customers. We should also make sure it's possible to directly create the `SharedCredentialsCache` without using `aws-config` (would be good to include as an example for the changelog).

Reply:
Thank you for pointing it out. That's certainly another place causing a breaking change. Provided the change log and an example in 9b4c55f.